src/modules/Game.class.js

```diff
@@ -1,68 +1,301 @@
 'use strict';
 
-/**
- * This class represents the game.
- * Now it has a basic structure, that is needed for testing.
- * Feel free to add more props and methods if needed.
- */
 class Game {
-  /**
-   * Creates a new game instance.
-   *
-   * @param {number[][]} initialState
-   * The initial state of the board.
-   * @default
-   * [[0, 0, 0, 0],
-   *  [0, 0, 0, 0],
-   *  [0, 0, 0, 0],
-   *  [0, 0, 0, 0]]
-   *
-   * If passed, the board will be initialized with the provided
-   * initial state.
-   */
-  constructor(initialState) {
+  static STATUS = {
+    IDLE: 'idle',
+    PLAYING: 'playing',
+    WIN: 'win',
+    LOSE: 'lose',
+  };
+
+  constructor(
+    initialState = [
+      [0, 0, 0, 0],
+      [0, 0, 0, 0],
+      [0, 0, 0, 0],
+      [0, 0, 0, 0],
+    ],
+  ) {
     // eslint-disable-next-line no-console
-    console.log(initialState);
+    this.status = Game.STATUS.IDLE;
+    this.initialState = initialState;
+    this.state = initialState.map((row) => [...row]);
+    this.score = 0;
+  }
+
+  getScore() {
+    return this.score;
+  }
+
+  getState() {
+    return this.state;
+  }
+
+  getStatus() {
+    return this.status;
+  }
+
+  start() {
+    this.status = Game.STATUS.PLAYING;
+    this.state = this.initialState.map((row) => [...row]);
+    this.addNewNumber();
+    this.addNewNumber();
+  }
+
+  restart() {
+    this.status = Game.STATUS.IDLE;
+    this.state = this.initialState.map((row) => [...row]);
+    this.score = 0;
+  }
+
+  addNewNumber() {
+    const emptyPositions = this.getEmptyPosition();
+
+    if (emptyPositions.length === 0) {
+      return;
+    }
+
+    const randomIndex = Math.floor(Math.random() * emptyPositions.length);
+    const [row, col] = emptyPositions[randomIndex];
+
+    this.state[row][col] = Math.random() < 0.9 ? 2 : 4;
+  }
+
+  getEmptyPosition() {
+    const emptyPositions = [];
+
+    for (let row = 0; row < 4; row++) {
+      for (let col = 0; col < 4; col++) {
+        if (this.state[row][col] === 0) {
+          emptyPositions.push([row, col]);
+        }
+      }
+    }
+
+    return emptyPositions;
+  }
+
+  moveLeft() {
+    if (this.getStatus() === Game.STATUS.PLAYING) {
+      const size = this.state.length;
+      const state = this.getState();
+      let moved = false;
+
+      for (let row = 0; row < size; row++) {
+        const newRow = [];
+        const merged = Array(size).fill(false);
+
+        for (let col = 0; col < size; col++) {
+          if (state[row][col] !== 0) {
+            newRow.push(state[row][col]);
+          }
+        }
+
+        for (let i = 0; i < newRow.length - 1; i++) {
+          if (newRow[i] === newRow[i + 1] && !merged[i] && !merged[i + 1]) {
+            newRow[i] *= 2;
+            this.score += newRow[i];
+            newRow.splice(i + 1, 1);
+            merged[i] = true;
+          }
```

Comment:
The logic for merging tiles in the 'moveLeft' method appears to be correct, but it's duplicated in the 'moveRight' method. Consider abstracting this logic into a separate method to follow the DRY (Don't Repeat Yourself) principle.